### PR TITLE
feat: tip NFT

### LINKED ISSUE #6022: [Crash] Error: Path parameter address cannot be an empty string.
I was trying to tip when I have no wallet connected, then Mask reports an error.

> Path parameter address cannot be an empty string.

Error stack:

<pre>Error: Path parameter address cannot be an empty string.
    at chrome-extension://cdiogojknfofcbengaanphjilhkhkiad/js/6645.js:38712:19
    at RegExp.[Symbol.replace] (<anonymous>)
    at String.replace (<anonymous>)
    at path (chrome-extension://cdiogojknfofcbengaanphjilhkhkiad/js/6645.js:38699:33)
    at urlcatImpl (chrome-extension://cdiogojknfofcbengaanphjilhkhkiad/js/6645.js:38652:14)
    at urlcat (chrome-extension://cdiogojknfofcbengaanphjilhkhkiad/js/6645.js:38642:16)
    at Object.resolveAddressLinkOnExplorer [as resolveAddressLink] (chrome-extension://cdiogojknfofcbengaanphjilhkhkiad/js/7335.js:2451:57)
    at TipDialog (chrome-extension://cdiogojknfofcbengaanphjilhkhkiad/js/chunk.9885.js:2770:171)
    at Nh (chrome-extension://cdiogojknfofcbengaanphjilhkhkiad/js/npm.react-dom.js:172:137)
    at Lk (chrome-extension://cdiogojknfofcbengaanphjilhkhkiad/js/npm.react-dom.js:295:386)</pre>

## Build info
- Version: 2.6.0(feat/tip-nft)
- NODE_ENV: production
- target: chromium
- build: beta
- architecture: web
- BUILD_DATE: 2022-04-02T15:31:01.963Z
- VERSION: v1.29.12-2185-g918d6a00e

## Git (*):

918d6a00e (feat/tip-nft) on tag "v2.5.0"
git@github.com:DimensionDev/Maskbook.git

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-337
https://mask.atlassian.net/browse/MF-334

image

close https://github.com/DimensionDev/Maskbook/issues/6022

### CODE REVIEW

File: packages/dashboard/src/pages/Wallets/components/CollectibleList/index.tsx [hunk 1]
```diff
@@ -64,6 +64,14 @@ export const CollectibleList = memo<CollectibleListProps>(({ selectedNetwork })
             Asset?.getNonFungibleAssets?.(account, { page: page, size: 20 }, undefined, selectedNetwork || undefined),
         [account, Asset?.getNonFungibleAssets, network, selectedNetwork],
     )
+    useEffect(() => {
```
Comment (UncleBill) [inline packages/dashboard/src/pages/Wallets/components/CollectibleList/index.tsx:67]: Put it close to `Asset.getNonFungibleAssets` since they are related.

**Comment by wenluomask** [comment]
Checked.

File: packages/mask/src/plugins/NextID/components/Tip/AddDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    const [tokenId, setTokenId] = useState('')
+    const t = useI18N()
+    const allNetworks = useNetworkDescriptors()
+    const network = useMemo(() => allNetworks.find((n) => n.chainId === chainId), [allNetworks, chainId])
+    const contract = useERC721TokenContract(contractAddress)
+    const { value: isMine, loading: isChecking } = useAsync(async () => {
+        console.log('contract', contract)
+        if (!contract || !account) return false
+        const ownerAddress = await contract.methods.ownerOf(tokenId).call()
+        console.log('ownerAddress', ownerAddress)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/NextID/components/Tip/AddDialog.tsx]:
```suggestion
        if (!contract || !account) return false
        const ownerAddress = await contract.methods.ownerOf(tokenId).call()
```

File: packages/mask/src/plugins/NextID/components/Tip/NFTSection/index.tsx [hunk 3]
```diff
@@ -59,71 +44,71 @@ const useStyles = makeStyles()((theme) => ({
...
     const { classes } = useStyles()
     const account = useAccount()
-    const { tokenDetailedOwnerList: myTokens = [] } = useERC721TokenDetailedOwnerList(erc721Contract, account)
-
     const selectedIds = useMemo(() => (erc721TokenId ? [erc721TokenId] : []), [erc721TokenId])
+    const { Asset } = useWeb3PluginState()
+
+    const networkDescriptor = useNetworkDescriptor()
+
+    // Can't not get the loading status of fetching via websocket
```
Comment (guanbinrui) [inline packages/mask/src/plugins/NextID/components/Tip/NFTSection/index.tsx]:
```suggestion
    // Cannot get the loading status of fetching via websocket
```

File: packages/web3-shared/evm/utils/transaction.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ export function isFinalState(type: TransactionStateType) {
 }
 
 /**
- * UNKNOWN -> WAIT_FOR_CONFIRMING
+ * UNKNOWN, FAILED -> WAIT_FOR_CONFIRMING
```
Comment (guanbinrui) [inline packages/web3-shared/evm/utils/transaction.ts]: Against the fact: A `failed` tx cannot recover to `wait_for_confirming`.

**Review by guanbinrui — DISMISSED** [review]

**Comment by wenluomask** [comment]
Checked.

**Review by guanbinrui — APPROVED** [review]

**Review by nuanyang233 — APPROVED** [review]

**Pull request merged**